src/api/CMakeLists.txt

```diff
@@ -40,25 +37,60 @@ PUBLIC ${PUBLIC_SOURCES_TO_ADD}
 set(OBJECTS_TO_ADD $<TARGET_OBJECTS:api> ${OBJECTS_TO_ADD})
 
 if (BUILD_SHARED_LIBS)
-  add_library(api_shared SHARED ${OBJECTS_TO_ADD})
-  target_link_libraries(api_shared ${GMP_LIBRARIES} ${GMPXX_LIBRARIES} Threads::Threads)
-  set_target_properties(api_shared PROPERTIES 
-      OUTPUT_NAME opensmt
-      VERSION ${OPENSMT_VERSION}
-      LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    add_library(OpenSMT SHARED ${OBJECTS_TO_ADD})
+    set_target_properties(OpenSMT PROPERTIES
+        NO_SYSTEM_FROM_IMPORTED true
+        VERSION ${OPENSMT_VERSION}
+        OUTPUT_NAME opensmt
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+      )
+
+  target_link_libraries(OpenSMT PUBLIC GMP::GMP Threads::Threads)
+  install(TARGETS OpenSMT EXPORT OpenSMTTargets
+      LIBRARY DESTINATION lib
+      ARCHIVE DESTINATION lib
+      RUNTIME DESTINATION bin
+      INCLUDES DESTINATION include
       )
-  install(TARGETS api_shared LIBRARY DESTINATION lib)
+
+  install(EXPORT OpenSMTTargets
+      FILE OpenSMTTargets.cmake
+      NAMESPACE OpenSMT::
+      DESTINATION lib/cmake/opensmt
+      )
+
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file( "${CMAKE_CURRENT_BINARY_DIR}/OpenSMTConfigVersion.cmake"
+    VERSION ${OPENSMT_VERSION}
+    COMPATIBILITY SameMajorVersion # SameMinorVersion available only from CMake 3.11 
+    )
+
+configure_package_config_file(OpenSMTConfig.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/OpenSMTConfig.cmake"
+    INSTALL_DESTINATION lib/cmake/opensmt
+    #PATH_VARS INCLUDE_INSTALL_DIR INSTALL_DIR
```

Comment:
Consider removing unless this is useful once we drop support for legacy cmake.

Reply:
I don't remember, most likely it can be removed.